### DISCUSSION TITLE
Make PublicFile.content more flexible than CID<IPFSUnixFSFile>

### DISCUSSION THREAD

**Discussion by fabricedesre** [wnfs-wg/spec#49] [General]
This was discussed in https://github.com/wnfs-wg/spec/discussions/18#discussioncomment-3324505 but it's not reflected in the spec. The MetaData field looks flexible enough but that is not matched by the content itself. I've started to look at mapping Capyloon's vfs resources to wnfs and with the current spec things will be quite ugly and/or inefficient.

I would need for content to be something like `CBOR<Record<string, CID>>` : being able to point to each resource variant by name. I'm not sure how that would fit with the private filesystem where the expectation will be to encrypt not just `content` but also the referenced CIDs.

**Comment by matheus23** [comment]
I've discussed this with Fabrice in discord, and I actually think that his use cases may be better served by putting more files into a file's metadata.
We should add dedicated support for reading "external" data out of metadata though, so I've added an issue to track that: #50
